_posts/2021-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,323 @@
+---
+layout: post
+title: Updating Foreman inventory with system facts
+date: 2021-01-12 17:00:00
+author: Marek Hulan
+tags:
+- foreman
+- community
+- tutorial
+- howto
+- beginners
+---
+
+This is a continuation of [How to start with Foreman post](https://theforeman.org/2020/12/how-to-start-with-foreman.html).
+Today we'll look at how the inventory can be kept up to date by receiving facts from hosts regularly. Then we'll take a look
+where facts can be used. We'll illustrate the configuration on one config management system, but the same concept is applicable
+to other facts sources. Links to the related documentation will be provided.
+
+<!--more-->
+
+## Automatically update the inventory from configuration management
+
+Having to keep a big inventory up to date manually would be a lot of work. Foreman
+can receive updates from other tools used to manage the infrastructure and reflect
+them in the database. Also new Hosts can be created this way. If you use some
+configuration management system or you use a subscription manager (mostly RPM
+based distributions), you can configure it to forward data about managed hosts to Foreman. If the host does not exist in Foreman yet, it can be automatically created. If it exists, the record is updated accordingly.
+
+Typically every configuration management agent run starts with gathering facts.
+Facts are generic information about the machine, OS, networking and other areas.
+These facts can be uploaded to Foreman. They are parsed and processed, so that Foreman knows e.g.
+about the current operating system version, kernel, IP addresses, available RAM etc.
+Foreman can also update other existing objects, such as associated subnet, OS or similar, if configured to do so. Also all the raw values
+are stored for later use. We simply call them Facts.
+
+## Facts importing overview
+
+The facts upload process is similar for all fact sources. The data is collected on the Smart Proxy machine and then sent to Foreman API. The request is authenticated using an x509 certificate. For the Authorization, the Foreman verifies, it comes from a Smart Proxy host, that have related Feature enabled (e.g. Puppet, Ansible etc). Then it imports the data and stores it to the database.
```

Comment:
```suggestion
The facts upload process is similar for all fact sources. The data is collected on the Smart Proxy machine and then sent to the Foreman API. The request is authenticated using an x509 certificate. For the authorization, the Foreman verifies the request comes from a Smart Proxy host that has related Feature enabled (e.g. Puppet, Ansible etc). Then Foreman imports the data and stores it to the database.
```

Reply:
This paragraph implies that the Smart Proxy does the collecting (the reaching out and collecting). Perhaps clarify that it's the tooling or services (e.g. Ansible, puppet, etc.) that send the data to the smart proxy which collects and sends it.

Reply:
reformulated it a bit, so it's hopefully clearer